### PR TITLE
1.14 sync with develop

### PR DESCRIPTION
**Pull request by jhendersonHDF**
Please look over this and let me know if there's anything that shouldn't be merged into the 1.14 branch.

@byrnHDF it was easiest to sync everything from develop, including the gzip and szip changes. If those weren't supposed to come to 1.14, please let me know so I can revert whichever commits are relevant.

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -74,28 +70,28 @@
...
     },
     {
       "name": "ci-ExamplesVars",
       "hidden": true,
       "cacheVariables": {
-        "HDF5_EXAMPLES_COMPRESSED": {"type": "STRING", "value": "hdf5-examples-2.0.4.tar.gz"},
-        "HDF5_EXAMPLES_COMPRESSED_DIR": {"type": "STRING", "value": "${sourceParentDir}/temp"},
-        "EXAMPLES_TGZ_ORIGPATH": {"type": "STRING", "value": "https://github.com/HDFGroup/hdf5-examples/archive/refs/tags/"},
-        "EXAMPLES_TGZ_ORIGNAME": {"type": "STRING", "value": "2.0.4.tar.gz"}
+        "HDF5_EXAMPLES_COMPRESSED": {"type": "STRING", "value": "hdf5-examples-master.tar.gz"},
```
Comment (jhendersonHDF) [inline CMakePresets.json:91]: Some of these examples links seemed suspect going from `2.0.4` to `master` so I didn't know if this was correct or not for 1.14 as opposed to develop

File: .github/workflows/h5py.yml
Comment (byrnHDF) [inline .github/workflows/h5py.yml:1]: This should not be added - cron type CI only works on default branch

File: .github/workflows/autotools.yml
Comment (jhendersonHDF) [inline .github/workflows/autotools.yml:1]: Looks like some of these synced workflow files need updating in this PR

File: .github/workflows/main-cmake.yml [hunk 2]
```diff
@@ -0,0 +1,251 @@
+name: hdf5 dev CMake CI
```
Comment (byrnHDF) [inline .github/workflows/main-cmake.yml]: name should be hdf5 1.14 in all these yaml files

File: .github/workflows/scorecard.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+name: Scorecard supply-chain security
+on:
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule:
+  # To guarantee Maintained check is occasionally updated. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
+  schedule:
+    - cron: '36 22 * * 4'
```
Comment (byrnHDF) [inline .github/workflows/scorecard.yml]: Should just remove the schedule item and leave the push?

File: CMakeFilters.cmake [hunk 4]
```diff
@@ -131,7 +119,7 @@ endif ()
 #-----------------------------------------------------------------------------
 # Option for SzLib support
 #-----------------------------------------------------------------------------
-option (HDF5_ENABLE_SZIP_SUPPORT "Use SZip Filter" OFF)
+option (HDF5_ENABLE_SZIP_SUPPORT "Use SZip Filter" ON)
```
Comment (byrnHDF) [inline CMakeFilters.cmake:122]: This should probably remain as previously

File: CMakePlugins.cmake
Comment (byrnHDF) [inline CMakePlugins.cmake:1]: This has an issue with blosc plugin - either the plugin stuff should be left as is or blosc will need to be excluded.

File: CMakePresets.json (hunk 1 above)
Comment (byrnHDF) [reply CMakePresets.json:91]: This will be gone when examples is moved in

File: .github/workflows/main-cmake.yml (hunk 2 above)
Comment (jhendersonHDF) [reply .github/workflows/main-cmake.yml]: will do

File: .github/workflows/scorecard.yml (hunk 3 above)
Comment (jhendersonHDF) [reply .github/workflows/scorecard.yml]: yes I think that may have been how it was before. Will revert

File: CMakeFilters.cmake (hunk 4 above)
Comment (jhendersonHDF) [reply CMakeFilters.cmake:122]: Was this change part of other changes or by itself? I can revert the commit or make the individual change; just concerned about getting confused between the state of this between develop and 1.14

File: config/cmake/ConfigureChecks.cmake [hunk 5]
```diff
@@ -916,7 +909,7 @@ endmacro ()
 #-----------------------------------------------------------------------------
 
 # ----------------------------------------------------------------------
-# Set the flag to indicate that the machine is using a special algorithm to convert
+# Set the flag to indicate that the machine is using a special algorithm toconvert
```
Comment (byrnHDF) [inline config/cmake/ConfigureChecks.cmake]: This went the wrong way!

File: CMakeFilters.cmake (hunk 4 above)
Comment (byrnHDF) [reply CMakeFilters.cmake:122]: There were other changes

**Comment by byrnHDF** [comment]
I'm not sure this is the way to do this - Reviewing will take a long time (it is slow on my computer!)
Maybe we should take it smaller bites?

File: config/cmake/ConfigureChecks.cmake (hunk 5 above)
Comment (jhendersonHDF) [reply config/cmake/ConfigureChecks.cmake]: Yes, looks like develop is wrong

**Comment by jhendersonHDF** [comment]
> I'm not sure this is the way to do this - Reviewing will take a long time (it is slow on my computer!) Maybe we should take it smaller bites?

The best way of course would be to keep the two branches in sync regularly. Since there are apparently 540 commits not merged from develop, I don't know the best way to go about this. Smaller batches may help, but there are still a ton of changes. Cherry picking out the tools move PR may help, but that will make it harder to do a regular merge of the other files.

**Comment by jhendersonHDF** [comment]
git merge-base shows that the last common commit between develop and the 1.14 branch is `b9f4c001ade4c230f8325080f3dd8181648324ab` from December 9. I could start there and pick up things in batches of a hundred commits or so, but the PRs that capture the tools move change and the `FALSE` -> `false`, `hbool_t` -> `bool` changes will still be huge.

**Review by lrknox — APPROVED** [review]

**Pull request merged**